homeassistant/components/whirlpool/climate.py

```diff
@@ -0,0 +1,189 @@
+        await self._aircon.connect()
+
+        try:
+            name = await self._aircon.fetch_name()
+            if name is not None:
+                self._attr_name = name
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.exception("Failed to get name")
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._aircon.get_online()
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._aircon.get_current_temp()
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._aircon.get_temp()
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        await self._aircon.set_temp(kwargs.get(ATTR_TEMPERATURE))
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self._aircon.get_current_humidity()
+
+    @property
+    def target_humidity(self):
+        """Return the humidity we try to reach."""
+        return self._aircon.get_humidity()
+
+    async def async_set_humidity(self, humidity: int) -> None:
+        """Set new target humidity."""
+        await self._aircon.set_humidity(humidity)
+
+    @property
+    def hvac_mode(self):
+        """Return current operation ie. heat, cool, fan."""
+        if not self._aircon.get_power_on():
+            return HVAC_MODE_OFF
+
+        mode: AirconMode = self._aircon.get_mode()
+        return AIRCON_MODE_MAP.get(mode, None)
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set HVAC mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            await self._aircon.set_power_on(False)
+            return
+
+        mode = HVAC_MODE_TO_AIRCON_MODE.get(hvac_mode)
+        if not mode:
+            _LOGGER.warning("Unexpected hvac mode")
+            return
+
+        await self._aircon.set_mode(mode)
+        if not self._aircon.get_power_on():
+            await self._aircon.set_power_on(True)
+
+    @property
+    def fan_mode(self):
+        """Return the fan setting."""
+        fanspeed = self._aircon.get_fanspeed()
+        return AIRCON_FANSPEED_MAP.get(fanspeed, FAN_OFF)
+
+    async def async_set_fan_mode(self, fan_mode):
+        """Set fan mode."""
+        fanspeed = FAN_MODE_TO_AIRCON_FANSPEED.get(fan_mode)
+        if not fanspeed:
+            return
+        await self._aircon.set_fanspeed(fanspeed)
+
+    @property
+    def swing_mode(self):
+        """Return the swing setting."""
+        return SWING_HORIZONTAL if self._aircon.get_h_louver_swing() else SWING_OFF
+
+    async def async_set_swing_mode(self, swing_mode):
+        """Set new target temperature."""
+        if swing_mode == SWING_HORIZONTAL:
+            await self._aircon.set_h_louver_swing(True)
+        else:
+            await self._aircon.set_h_louver_swing(False)
```

Comment:
```suggestion
        await self._aircon.set_h_louver_swing(swing_mode == SWING_HORIZONTAL)
```

Not sure if saves much, but you could write this like the above

Reply:
Duh! :smile: